ethcore/wasm/src/runtime.rs

```diff
@@ -369,13 +412,17 @@ impl<'a, 'b> Runtime<'a, 'b> {
 		-> Result<Option<interpreter::RuntimeValue>, InterpreterError>
 	{
 		let amount = context.value_stack.pop_as::<i32>()? as u32;
+
+		charge!(self, self.ext.schedule().wasm.alloc * amount);
```

Comment:
depending on the alloc cost, this seems like it could easily overflow a u32 when allocating just a few MB.

Reply:
it's super easy operation with arena allocator, so it should not be priced more than memory opcodes
and memory limit for wasm is just 16mb

i will change to u64 anyway though to be uniform